opal/verify.go

```diff
@@ -35,6 +38,62 @@ func validateReviewerConfigDuringCreate(d *schema.ResourceData) error {
 			return nil
 		}
 	}
+	if requestConfigurationListI, ok := d.GetOkExists("request_configuration"); ok {
+		if len(requestConfigurationListI.([]interface{})) > 0 {
+			return nil
+		}
+	}
 
 	return errors.New("Invalid reviewer configuration. Please specify at least 1 reviewer stage, or set auto_approval to true or set is_requestable to false")
 }
+
+func validateRequestConfigurationListDuringCreate(ctx context.Context, d *schema.ResourceData) error {
+	_, autoApprovalOk := d.GetOkExists("auto_approval")
+	_, requireMfaToRequestOk := d.GetOkExists("require_mfa_to_request")
+	_, requireSupportTicketOk := d.GetOkExists("require_support_ticket")
+	_, maxDurationOk := d.GetOk("max_duration")
+	_, recommendedDurationOk := d.GetOk("recommended_duration")
+	_, requestTemplateIDOk := d.GetOk("request_template_id")
+	_, isRequestableOk := d.GetOk("is_requestable")
+	requestConfigurationListI, requestConfigurationListOk := d.GetOk("request_configuration")
+	oldRequestConfigurationFieldsChanged := autoApprovalOk || requireMfaToRequestOk || requireSupportTicketOk || maxDurationOk || recommendedDurationOk || requestTemplateIDOk || isRequestableOk
+
+	if requestConfigurationListOk {
+		if oldRequestConfigurationFieldsChanged {
+			return errors.New("Cannot set both request_configuration_list and any of auto_approval, require_mfa_to_request, require_support_ticket, is_requestable, max_duration, recommended_duration, or request_template_id.")
+		}
+		if len(requestConfigurationListI.([]interface{})) < 1 {
+			return errors.New("Invalid request configuration list. Please specify at least 1 request configuration")
```

Comment:
should there be a maximum, like with request stages?

Reply:
Oh hm i'll set it to 3 as well

Reply:
on second thought I think it should be fine? Maybe I can add that as an actual constraint eventually